File: api/src/main/java/io/github/wulkanowy/api/exams/Exam.java
package io.github.wulkanowy.api.exams;

@Deprecated
public class Exam {

    private String subjectAndGroup = "";

    private String type = "";

    private String description = "";

    private String teacher = "";

    private String entryDate = "";

    public String getSubjectAndGroup() {
        return subjectAndGroup;
    }

    public Exam setSubjectAndGroup(String subjectAndGroup) {
        this.subjectAndGroup = subjectAndGroup;
        return this;
    }

    public String getType() {
        return type;
    }

    public Exam setType(String type) {
        this.type = type;
        return this;
    }

    public String getDescription() {
        return description;
    }

    public Exam setDescription(String description) {
        this.description = description;
        return this;
    }

    public String getTeacher() {
        return teacher;
    }

    public Exam setTeacher(String teacher) {
        this.teacher = teacher;
        return this;
    }

    public String getEntryDate() {
        return entryDate;
    }

    public Exam setEntryDate(String entryDate) {
        this.entryDate = entryDate;
        return this;
    }
}

File: app/src/main/java/io/github/wulkanowy/data/db/entities/Exam.kt
package io.github.wulkanowy.data.db.entities

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import java.io.Serializable
import java.time.LocalDate

@Entity(tableName = "Exams")
data class Exam(

    @ColumnInfo(name = "student_id")
    val studentId: Int,

    @ColumnInfo(name = "diary_id")
    val diaryId: Int,

    val date: LocalDate,

    @ColumnInfo(name = "entry_date")
    val entryDate: LocalDate,

    val subject: String,

    @Deprecated("not available anymore")
    val group: String,

    val type: String,

    val description: String,

    val teacher: String,

    @ColumnInfo(name = "teacher_symbol")
    val teacherSymbol: String
) : Serializable {

    @PrimaryKey(autoGenerate = true)
    var id: Long = 0

    @ColumnInfo(name = "is_notified")
    var isNotified: Boolean = true
}
